pkg/apis/options/upstreams.go

```diff
@@ -0,0 +1,60 @@
+package options
+
+import "time"
+
+// Upstreams is a collection of definitions for upstream servers.
+type Upstreams []Upstream
+
+// Upstream represents the configuration for an upstream server.
+// Requests will be proxied to this upstream if the path matches the request path.
+type Upstream struct {
+	// ID should be a unique identifier for the upstream.
+	// This value is required for all upstreams.
+	ID string `json:"id"`
+
+	// Path is used to map requests to the upstream server.
+	// The closest match will take precedence and all Paths must be unique.
+	Path string `json:"path"`
+
+	// The URI of the upstream server. This may be an HTTP(S) server of a File
+	// based URL. It may include a path, in which case all requests will be served
+	// under that path.
+	// Eg:
+	// - http://localhost:8080
+	// - https://service.localhost
+	// - https://service.localhost/path
+	// - file://host/path
+	// If the URI's path is "/base" and the incoming request was for "/dir",
+	// the upstream request will be for "/base/dir".
+	URI string `json:"uri"`
+
+	// InsecureSkipTLSVerify will skip TLS verification of upstream HTTPS hosts.
+	// This option is insecure and will allow potential Man-In-The-Middle attacks
+	// betweem OAuth2 Proxy and the usptream server.
+	// Defaults to false.
+	InsecureSkipTLSVerify bool `json:"insecureSkipTLSVerify"`
```

Comment:
Now that alternate CA trust sources are possible, should we allow that to be handled here per upstream?

https://github.com/oauth2-proxy/oauth2-proxy/pull/645

Reply:
The referenced PR only affects the setting for the default http.Client, as far as I was aware the reverse proxies use their own http.Client and therefore aren't affected by that option. I don't think we have an option for the ability to specify that for upstreams yet do we? I'd prefer to add that in a later PR if we want that

Reply:
Ah yeah -- you're right.  This isn't the place to add that.  Let's punt (and maybe very long term, the original merits of #645 seemed very corner case-y to me -- with the ideal route being getting an appropriate system CA setup for environment requirements).